packages/headless/src/app/product-listing-engine/product-listing-engine.ts

```diff
@@ -66,10 +73,12 @@ export function buildProductListingEngine(
     options.configuration,
     logger
   );
+  const searchAPIClient = createSearchAPIClient(options.configuration, logger);
 
   const thunkArguments: ProductListingThunkExtraArguments = {
     ...buildThunkExtraArguments(options.configuration, logger),
     productListingClient,
+    searchAPIClient,
```

Comment:
This is a bit unusual, but I think it is required that we use the `searchAPIClient` even in product listings so that we can leverage facet search.

Our facets are pretty much the same thing as the standard facets, except that they pass through the Product Listing API instead of Search API.

So I think that this can work :smile:

Reply:
1- I did this to prevent rewriting all actions related to `executeFacetSearch`... they expect an `AsyncThunkSearchOptions`... It really seemed overkill to do a "core-facet-search" :(

2- I also don't think we *need* to provide a facet search specific to Product Listing since our facets work exactly the same way as the Search API. If so, *every* service will need to implement a facet search endpoint that all does the same thing :/ We could rediscuss this outside of this thread, but right now, I think it's cleaner to use the search's endpoint for that.

Reply:
> I also don't think we need to provide a facet search specific to Product Listing since our facets work exactly the same way as the Search API. If so, every service will need to implement a facet search endpoint that all does the same thing :/ We could rediscuss this outside of this thread, but right now, I think it's cleaner to use the search's endpoint for that.

Yeah I'd like to discuss that with you and other people, because it is true that it does add burden on us (Coveo as a company) to maintain "repetitive" stuff across different services.

However, let's remember that the original intention behind separating all use cases (Service, Commerce, Classic search, Workplace) behind different services is to try to make it easier/more sane for external developer that looks at our API and "understand" what they should do. 

Like, if we really wanted to optimize "removing as much workload for Coveo R&D", well... we would never have created different services for different uses cases in the first place :) We would have stuck with the current approach of one API (Search API) being used for everything.

Versus, if we want to optimize "a developer that knows nothing about Coveo, that looks at our API and tries to do a commerce deployment, should be able to somewhat understand what to do", it's a different thing. By definition, trying to make things simpler for developers building on our platform means * a TON * more work for us.

It's not black and white however: If everybody uses headless to do all commerce deployments (and never try to use API directly), then it's true that the complexity is "hidden".

I'd love to talk about this topic with as many people as possible !

Reply:
After discussion with Sami, we'll tackle facetSearch in another PR.

One idea that we pitched around was to create a common interface specifically for `facetSearch` and implement it in both searchApiClient and productListingClient.

This would work pretty well, but I'm not 100% it scales that much, we'll have to add an extra client reference in the thunk argument for each "common" methods that we need to implement. 

Which is why we'll test it out elsewhere and keep `searchApiClient` here for now :)